OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/fieldaccess/reflection/FieldMatcher.scala

```diff
@@ -0,0 +1,150 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package tac
+package fpcf
+package analyses
+package fieldaccess
+package reflection
+
+import org.opalj.br.ObjectType
+import org.opalj.br.analyses.ProjectIndexKey
+import org.opalj.br.analyses.SomeProject
+import org.opalj.br.Field
+import org.opalj.br.FieldType
+import org.opalj.tac.fpcf.analyses.cg.V
+import org.opalj.tac.fpcf.analyses.cg.isTypeCompatible
+import org.opalj.value.IsReferenceValue
+
+import scala.collection.immutable.ArraySeq
+
+/**
+ * Used to determine whether a certain field should be considered as a target for a reflective access.
+ * These accesses should be resolved by chaining matchers in [[FieldMatching.getPossibleFields]].
+ *
+ * @author Maximilian Rüsch
+ */
+trait FieldMatcher {
+
+    def initialFields(implicit p: SomeProject): Iterator[Field]
+    def contains(field: Field)(implicit p: SomeProject): Boolean
+    def priority: Int
+}
+
+final class NameBasedFieldMatcher(val possibleNames: Set[String]) extends FieldMatcher {
+
+    override def initialFields(implicit p: SomeProject): Iterator[Field] = {
+        possibleNames.iterator.flatMap(p.get(ProjectIndexKey).findFields)
+    }
+    override def contains(f: Field)(implicit p: SomeProject): Boolean = possibleNames.contains(f.name)
+    override def priority: Int = 2
+}
+
+class ClassBasedFieldMatcher(
+        val possibleClasses:          Set[ObjectType],
+        val onlyFieldsExactlyInClass: Boolean
+) extends FieldMatcher {
+
+    // TODO use a ProjectInformationKey or WeakHashMap to cache fields per project
+    // (for the contains check)
+    private[this] def fields(implicit p: SomeProject): Set[Field] = possibleClasses.flatMap { c =>
+        // todo what about "inherited" fields?
+        p.classFile(c).map(_.fields).getOrElse(ArraySeq.empty)
+    }
+
+    override def initialFields(implicit p: SomeProject): Iterator[Field] = fields.iterator
+    override def contains(f: Field)(implicit p: SomeProject): Boolean = fields.contains(f)
```

Comment:
I wonder if it would be more efficient to use the field's declaring class file here and check whether the possible classes contain it (or, if we consider inherited fields, any of them is a subclass)

Reply:
Should be obsolete as per https://github.com/opalj/opal/pull/145#discussion_r1381668717. We can discuss this in more depth in the follow-up PR.